src/analysis/asymptotics.lean

```diff
@@ -1058,6 +1058,113 @@ theorem is_o_iff_tendsto {f g : α → 𝕜} {l : filter α}
   is_o f g l ↔ tendsto (λ x, f x / (g x)) l (𝓝 0) :=
 iff.intro is_o.tendsto_0 (is_o_of_tendsto hgf)
 
+/-!
+### Eventually (u / v) * v = u
+
+If `u` and `v` are linked by an `is_O_with` relation, then we
+eventually have `(u / v) * v = u`, even if `v` vanishes.
+-/
+
+section eventually_mul_div_cancel
+
+variables {u v : α → 𝕜}
+
+lemma is_O_with.eventually_mul_div_cancel (h : is_O_with c u v l) :
+  ∀ᶠ x in l, (u x / v x) * v x = u x :=
+begin
+  rw is_O_with at h,
+  rw eventually_iff_exists_mem at *,
+  rcases h with ⟨s, hsl, hs⟩,
+  use [s, hsl],
+  intros y hy,
```

Comment:
Untested:
```suggestion
  refine h.mono (λ y hy, _),
```

Reply:
Then you can use [div_mul_cancel_of_imp](https://leanprover-community.github.io/mathlib_docs/algebra/group_with_zero.html#div_mul_cancel_of_imp).